source/Company.py

```diff
@@ -1,53 +1,184 @@
+# Company Class exercise
+
+def contains_number(value):
+    '''
+    A function which checks if a value includes digits
+    '''
+    if True in [char.isdigit() for char in value]:
+        return True
+    return False
+
+
+def check_string(str: str):
+    '''
+    A function which checks if the conditions of
+    company name are met
+    '''
+    if contains_number(str) or len(str) < 2:
+        return False
+    elif str[0].isupper() and len(str) >= 2:
+        return True
+
+
+def check_int(val):
+    '''
+    A function which checks if a given number is int
+    '''
+    if isinstance(val, int):
+        return True
+    return False
+
+
+def check_positive(num: float):
+    '''
+    A function which returns True if the given number
+    is positive and False otherwise
+    '''
+    if num <= 0:
+        return False
+    return True
+
 
 class Company:
     # you can add methods for your usage
 
-    def __init__(self, name, stocks_num, stock_price, comp_type):
-        pass
+    # The constructor function
+    def __init__(self, name: str, stocks_num: int,
+                 stock_price: float, comp_type: str):
+        if not check_string(name):
+            raise ValueError("Error! Company name starts with capital letter.")
+        else:
+            self.name = name
+        if check_positive(stocks_num) is False or check_int(stocks_num) is False:
+            raise ValueError("Error! Number of stocks must be positive.")
+        else:
+            self.stocks_num = stocks_num
+        if not check_positive(stock_price):
+            raise ValueError("Error! Price of stock must be positive.")
+        else:
+            self.stock_price = stock_price
+        if not check_string(comp_type):
+            raise ValueError("Error! Company type starts with capital letter.")
+        else:
+            self.comp_type = comp_type
```

Comment:
Constructor implemented almost perfectly! great!

Reply:
Now it is :)